src/sign.ts

```diff
@@ -75,32 +66,29 @@ export class Signer {
     // Pre-authentication encoding to be signed
     const paeBuffer = dsse.preAuthEncoding(payloadType, payload);
 
-    // Get signature and signing certificate for pae
-    const { signature, certificates } = await this.sign(paeBuffer);
+    // Get signature and verification material for pae
+    const sigMaterial = await this.signer(paeBuffer);
 
     const envelope: Envelope = {
       payloadType,
       payload: payload,
       signatures: [
         {
-          keyid: '',
-          sig: signature,
+          keyid: sigMaterial.key?.id || '',
+          sig: sigMaterial.signature,
         },
       ],
     };
 
-    const signingCertificate = certificates[0];
-
-    const proposedEntry = rekor.toProposedIntotoEntry(
-      envelope,
-      signingCertificate
-    );
+    const proposedEntry = rekor.toProposedIntotoEntry(envelope, sigMaterial);
     const entry = await this.rekor.createEntry(proposedEntry);
```

Comment:
Any thoughts on how we could get this proposed entry and not create the Rekor entry at the same time? The interface doesn't feel ideal, as it looks like we'd store the `proposedEntry` and `sigMaterial` to later create the entry and the dsse bundle 🤔